src/logic/safe/transactions/gas.ts

```diff
@@ -29,74 +23,75 @@ const estimateDataGasCosts = (data: string): number => {
-  safeAddress: string,
-  to: string,
-  data: string,
-  tx?: Transaction,
-  preApprovingOwner?: string,
-): Promise<number> => {
-  try {
-    const web3 = getWeb3()
-    const from = await getAccountFrom(web3)
-
-    if (!from) {
-      return 0
-    }
+// https://docs.gnosis.io/safe/docs/contracts_signatures/#pre-validated-signatures
+export const getPreValidatedSignatures = (from: string): string => {
+  return `0x000000000000000000000000${from.replace(
+    EMPTY_DATA,
+    '',
+  )}000000000000000000000000000000000000000000000000000000000000000001`
+}
 
-    const safeInstance = (new web3.eth.Contract(GnosisSafeSol.abi as AbiItem[], safeAddress) as unknown) as GnosisSafe
-    const nonce = await safeInstance.methods.nonce().call()
-    const threshold = await safeInstance.methods.getThreshold().call()
-    const isExecution = tx?.confirmations.size === Number(threshold) || !!preApprovingOwner || threshold === '1'
-
-    let txData
-    if (isExecution) {
-      // https://docs.gnosis.io/safe/docs/docs5/#pre-validated-signatures
-      const signatures = tx?.confirmations
-        ? generateSignaturesFromTxConfirmations(tx.confirmations, preApprovingOwner)
-        : `0x000000000000000000000000${from.replace(
-            EMPTY_DATA,
-            '',
-          )}000000000000000000000000000000000000000000000000000000000000000001`
-      txData = await safeInstance.methods
-        .execTransaction(
-          to,
-          tx?.value || 0,
-          data,
-          CALL,
-          tx?.safeTxGas || 0,
-          0,
-          0,
-          ZERO_ADDRESS,
-          ZERO_ADDRESS,
-          signatures,
-        )
-        .encodeABI()
-    } else {
-      const txHash = await safeInstance.methods
-        .getTransactionHash(to, tx?.value || 0, data, CALL, 0, 0, 0, ZERO_ADDRESS, ZERO_ADDRESS, nonce)
-        .call({
-          from,
-        })
-      txData = await safeInstance.methods.approveHash(txHash).encodeABI()
-    }
+export type TransactionEstimationProps = {
+  txData: string
+  safeAddress: string
+  txRecipient: string
+  isExecution: boolean
+  txAmount?: string
+  operation?: number
+}
 
-    const gas = await calculateGasOf(txData, from, safeAddress)
-    const gasPrice = await calculateGasPrice()
+export const estimateTransactionGas = async ({
+  txAmount,
+  txData,
+  txRecipient,
+  isExecution,
+  safeAddress,
+  operation = CALL,
+}: TransactionEstimationProps): Promise<number> => {
+  const web3 = getWeb3()
+  const from = await getAccountFrom(web3)
 
-    return gas * parseInt(gasPrice, 10)
-  } catch (err) {
-    console.error('Error while estimating transaction execution gas costs:')
-    console.error(err)
+  if (!from) {
+    return 0
+  }
 
-    return 10000
+  if (isExecution) {
+    // Gas of executing a transaction within the safe (threshold reached and transaction ready to be executed)
+    return await estimateExecTransactionGas(safeAddress, txData, txRecipient, txAmount || '0', operation)
```

Comment:
When you are returning a Promise outside a try/catch block you can return just the Promise without awaiting.

Reply:
You are right I removed them thanks 👍